stixcore/util/scripts/end2end_testing.py

```diff
@@ -0,0 +1,150 @@
+from stixcore.processing.L0toL1 import Level1
+from stixcore.processing.LBtoL0 import Level0
+from stixcore.processing.TMTCtoLB import process_tmtc_to_levelbinary
+from stixcore.products.product import Product
+from stixcore.tmtc.packets import TMTC
+
+pytest.skip
+
+END_TO_END_TEST_FILES = [
+    # L1
+    # science
+    "L1/2021/06/28/SCI/solo_L1_stix-sci-xray-rpd_20210628T092301_20210628T092501_V01_2106280010-54759.fits", # noqa
+    "L1/2021/06/28/SCI/solo_L1_stix-sci-xray-cpd_20210628T190505_20210628T191500_V01_2106280009-54755.fits", # noqa
+    "L1/2021/06/28/SCI/solo_L1_stix-sci-xray-scpd_20210628T092301_20210628T092502_V01_2106280006-54720.fits", # noqa
+    "L1/2021/06/28/SCI/solo_L1_stix-sci-xray-vis_20210628T092301_20210628T092502_V01_2106280004-54716.fits", # noqa
+    "L1/2021/10/13/SCI/solo_L1_stix-sci-aspect-burst_20211013T034959_20211013T035842_V01_0.fits", # noqa
+    "L1/2021/06/28/SCI/solo_L1_stix-sci-xray-spec_20210628T230112_20210628T234143_V01_2106280041-54988.fits", # noqa
+    # QL
+    "L1/2020/06/16/QL/solo_L1_stix-ql-background_20200616_V01.fits",
+    "L1/2020/06/16/QL/solo_L1_stix-ql-flareflag_20200616_V01.fits",
+    "L1/2020/06/16/QL/solo_L1_stix-ql-lightcurve_20200616_V01.fits",
+    "L1/2020/06/16/QL/solo_L1_stix-ql-variance_20200616_V01.fits",
+    "L1/2021/11/16/QL/solo_L1_stix-ql-spectra_20211116_V01.fits",
+    "L1/2021/11/16/CAL/solo_L1_stix-cal-energy_20211116_V01.fits",
+    # HK
+    "L1/2020/06/16/HK/solo_L1_stix-hk-maxi_20200616_V01.fits",
+    "L1/2021/09/20/HK/solo_L1_stix-hk-mini_20210920_V01.fits"]
+
+remote = ["http://pub099.cs.technik.fhnw.ch/data/fits_test/" + x for x in END_TO_END_TEST_FILES]
+# files = ["/home/shane/fits_test/" + x for x in files]
+files = [("/home/shane/fits_test_latest/" + x, remote[i])
+         for i, x in enumerate(END_TO_END_TEST_FILES)]
+
+
+def _pretty_xml_print(current, parent=None, index=-1, depth=0):
+    for i, node in enumerate(current):
+        _pretty_xml_print(node, current, i, depth + 1)
+    if parent is not None:
+        if index == 0:
+            parent.text = '\n' + ('\t' * depth)
+        else:
+            parent[index - 1].tail = '\n' + ('\t' * depth)
+        if index == len(parent) - 1:
+            current.tail = '\n' + ('\t' * (depth - 1))
+
+
+def rebuild_end2end(files, *, splits=3, socdir=Path("/home/shane/tm/"),
+                    outdir=test_data.products.DIREND2END):
+    splitfiles = list(range(0, splits))
+    packetid = 0
+    openxml = ""
+    rootxml = None
+
+    outdir.mkdir(parents=True, exist_ok=True)
+
+    newroot = [Et.Element('Response') for i in splitfiles]
+    newresponse = [Et.SubElement(newroot[i], 'PktRawResponse') for i in splitfiles]
+
+    for f, u in files:
+        try:
+            l1_f = Path(f)
+            l1_p = Product(l1_f)
+            rootdir = l1_f.parent.parent.parent.parent.parent.parent
+
+            l0_f = (rootdir / "L0").rglob(l1_p.parent[0]).__next__()
```

Comment:
I don't think there is any guarantee of wording form `glob` so it may very from OS to OS of because  of access or modify date